src/software/amazon/ion/Timestamp.java

```diff
@@ -1854,18 +1944,37 @@ public void print(Appendable out)
         print(out, adjusted);
     }
 
+    /**
+     * Safe version of print failing if the fractional second scale is bigger than a safe threshold to
```

Comment:
I'm surprised this doesn't truncate the output rather than fail.  That seems more useful IMO.  Put another way, how would the user handle the failure?

Did you consider allowing the caller to provide the max precision, rather than hard-coding it?